### PR TITLE
Fix disabled attribute on checkout on classic

### LINKED ISSUE #26274: After upgrade, theme must be updated to be able to pay into the final step on PrestaShop 1.7.8
#### Describe the bug

Since PrestaShop 1.7.8, a change inside the core.js affect all themes and disallow the possibility of payments inside the checkout.

This issue was introduced with this pull request: https://github.com/PrestaShop/PrestaShop/pull/18904/

As now, the disabled thing is moved on class and not on attribute.
As like every theme have followed the Classic theme about that, if someone update it's PrestaShop without using Classic (and even without update CLassic, as it's possible), it will disallow the payment.

#### Expected behavior

We can add the disabled class, but the remain of disabled attribute into the JavaScript is needed.

#### Steps to Reproduce

1. Install PrestaShop 1.7.7.8
2. Update to PrestaShop 1.7.8.0 without updating Classic theme.
3. See error

**Screenshots**

If applicable, add screenshots or screen recordings to help explain your problem.

#### Additional information

* PrestaShop version: 1.7.8.0
* PHP version: N/A

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 1.7.8.x
| Description?      | The button is disabled even if everything is fine at the last step of checkout
| Type?             | bug fix
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #26274.
| How to test?      | See issue, keep in mind that you need to have an 1.7.7, then upgrade to 1.7.8 in order to reproduce the bug (and then test it)
| Possible impacts? | Checkout on FO

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/26289)

### CODE REVIEW

**Review by matks — DISMISSED** [review]
Thank you @NeOMakinG . Maybe we should add a comment in the code to help reader to understand why this line is needed?

**Comment by NeOMakinG** [comment]
> Thank you @NeOMakinG . Maybe we should add a comment in the code to help reader to understand why this line is needed?

Done

File: themes/_core/js/checkout-payment.js [hunk 1]
```diff
@@ -111,6 +111,8 @@ class Payment {
     } else {
       this.showConfirmation();
       $(`${this.confirmationSelector} button`).toggleClass('disabled', !show);
+      // We keep in order to be retro compatible with older themes
```
Comment (matks) [inline themes/_core/js/checkout-payment.js]:
```suggestion
      // Next line provides backward compatibility for Classic Theme < 1.7.8
```
'Retro' is French 😄

**Review by PrestaEdit — DISMISSED** [review]

**Review by ghost — APPROVED** [review]

**Comment by jf-viguier** [comment]
thanks @NeOMakinG

**Review by PierreRambaud — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by tegbessou — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @NeOMakinG,

It is ok :heavy_check_mark: 
I checked PS1780 after an upgrade from PS1778 using the 1-click upgrade v4.12.0
- Multishop enabled: for each shop => a custom theme is installed
**With PR**
image
**Without PR**
15643175-9fa0b1911c554bae695d5f758fb54a62

I attached a screen record
https://watch.screencastify.com/v/BHDizPpJtU72ftVB2pXQ

I checked fresh install of this PR with a different theme => ok (Checkout => ok)

I checked the upgrade from PS16124 to this PR with changing theme => ok (Checkout => ok)

I checked the upgrade from PS1777 to this PR using a different theme => ok (Checkout => ok)
I checked the upgrade from PS1780 to this PR => ok
image

Thanks!

**Comment by Progi1984** [comment]
Thanks @NeOMakinG & @khouloudbelguith

**Comment by jacadesing** [comment]
Hello, I cannot find the file to modify, in my prestashop the directory you indicate does not exist.

Can you help me?

Thank you

**Comment by NeOMakinG** [comment]
@jacadesing what you can do is remove the `disabled` attribute on the button, it can be a quick fix in your case, because you can't modify the source of the JS in production, and add the `.disabled` class

**Comment by jacadesing** [comment]
He actualizado ha prestashop 1.7.8.1 y el modulo de prestashop checkout método de pago por tarjeta sigue sin habilitarse.

I have updated prestashop 1.7.8.1 and the prestashop checkout card payment method is still not enabled.

**Comment by yannicknya** [comment]
I am in version 1.7.8.1 and the payment button is still disabled.
I migrated from 1.7.8.0 to 1.7.8.1 with the Click upgrade module


_moreover, on a payment in credit card, I have the impression that if one does not put the information (number of card, date, cvc), one can initiate a payment, it is normal behavior?_

**Comment by NeOMakinG** [comment]
Please guys, create an issue with a way to test it, I can't reproduce it myself, it should be specific to your project!

**Comment by jacadesing** [comment]
Hello yannicknya, it actually still doesn't work, the module still doesn't get an update to solve that bug.

I have half solved it, because there are users who still cannot pay, removing the smart fields from the form.

**Pull request merged**